PWGHF/HFL/Tasks/taskElectronWeakBoson.cxx

```diff
@@ -0,0 +1,244 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+///
+/// \ task for WeakBoson (W/Z) based on electron in mid-rapidity
+///        
+/// \S. Sakai & S. Ito
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Common/DataModel/EventSelection.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/DataModel/PIDResponse.h"
+#include "Framework/ASoAHelpers.h"
+
+#include "EMCALBase/Geometry.h"
+#include "EMCALCalib/BadChannelMap.h"
+#include "PWGJE/DataModel/EMCALClusters.h"
+#include "DataFormatsEMCAL/Cell.h"
+#include "DataFormatsEMCAL/Constants.h"
+#include "DataFormatsEMCAL/AnalysisCluster.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::framework::expressions;
+
+using selectedClusters = o2::soa::Filtered<o2::aod::EMCALClusters>;
+
+// PbPb
+using TrackEle = o2::soa::Filtered<o2::soa::Join<o2::aod::Tracks, o2::aod::FullTracks, o2::aod::TracksExtra_001, o2::aod::TracksDCA, o2::aod::TrackSelection, o2::aod::pidTPCFullEl>>; 
+
+// pp
+//using TrackEle = o2::soa::Filtered<o2::soa::Join<o2::aod::Tracks, o2::aod::FullTracks, o2::aod::TracksDCA, o2::aod::TrackSelection, o2::aod::pidTPCEl, o2::aod::pidTOFEl>>; 
+
+struct HfTaskElectronWeakBoson {
+
+  Preslice<o2::aod::EMCALClusterCells> perCluster = o2::aod::emcalclustercell::emcalclusterId;
+  Preslice<o2::aod::EMCALAmbiguousClusterCells> perClusterAmb = o2::aod::emcalclustercell::emcalambiguousclusterId;
+  PresliceUnsorted<o2::aod::EMCALMatchedTracks> perClusterMatchedTracks = o2::aod::emcalmatchedtrack::trackId;
+
+  Configurable<int> nBinsPt{"nBinsPt", 100, "N bins in pt histo"};
+  Configurable<int> nBinsE{"nBinsE", 100, "N bins in E histo"};
+
+  // event filter
+  Filter eventFilter = (o2::aod::evsel::sel8 == true);
+  Configurable<float> vtxZ{"vtxZ", 10.f, ""};
+  Filter posZFilter = (nabs(o2::aod::collision::posZ) < vtxZ);
+
+  // track cuts
+  Filter filter_globalTr = requireGlobalTrackInFilter();  
+
+  Configurable<float> etalow{"etalow", -0.6f, ""};
+  Configurable<float> etaup{"etaup", 0.6f, ""};
```

Comment:
Pay attention to proper names.

Reply:
keep as it is

Reply:
Please don't mark comments as resolved when you did not address them. The current names are ambiguous (eta of what? what is the meaning of up/low?), they don't follow naming conventions and the variables are not documented. Please fix it.